src/daggy.js

```diff
@@ -169,6 +171,13 @@
     }
   }
 
+  function makeConstructorFromObject(fields) {
+    return function(obj) {
+      var values = fields.map(function(field) { return obj[field]; });
+      return this.apply(null, values);
```

Comment:
I think you could pass `proto` to this function too (the one which `makeConstructor` accepts) and then use:
`makeValue(fields, proto, values, values.length);` instead of `return this.apply(null, values);`